articles/big-o-notation/index.md

```diff
@@ -0,0 +1,223 @@
+### Understanding the Big O notation
+
+![bigoh](/engineering-education/big-o-notation/bigo.png)
+
+We all need a way to measure the performance (efficiency) of our algorithms as they scale up.
```

Comment:
It would be great if you mention why we choose time and space to measure complexity. 

For example,
### Factors on which running time of a program may depend on.

- Single Processor v Multi Processor
- CPU Speed
- Read and Write Speed
- Configuration of Machine
- Input Size

    **Since most of these factors are all machine-dependent, we need a universal metric to evaluate running time**

    Hence, we measure running time of a program as a **function of its input size.**

Reply:
Noted

Reply:
Done